dpnp/dpnp_iface_manipulation.py

```diff
@@ -89,18 +89,13 @@ def asfarray(a, dtype=None):
 
     """
```

Comment:
please update tests also:
- remove `allow_fall_back_on_numpy` from `tests/test_arraymanipulation.py::test_asfarray`
- extend `tests/test_sycl_queue.py::test_array_copy` and `tests/test_usm_type.py::test_array_copy`
- add `tests/cupy_tests/manipulation_tests/test_kind.py`